pyproject.toml

```diff
@@ -61,10 +61,13 @@ ipython = [
 ]
 jupyter = [  # these are optional dependencies for utils.console and table
     "astropy[ipython]",
-    "ipywidgets",
-    "ipykernel",
-    "ipydatagrid",
-    "pandas",
+    "ipywidgets>=7.7.3",
+    "ipykernel>=6.16.0",
+    "ipydatagrid>=1.1.13",
+    # jupyter-core is a transitive dependency via ipykernel, we declare it as
+    # a direct dependency in order to set a lower bound for oldest-deps testing
+    "jupyter-core>=4.11.2",
```

Comment:
this is a perhaps non-ideal fix; this implies that `jupyter-core` will still be installed even in the (however unlikely) event that `ipykernel` stops depending on it. It might also linger if we ever remove `ipykernel` and forget about this line.
Another possibly cleaner fix would be to use a constraint file as described in https://tox.wiki/en/latest/config.html#deps, though maybe the cost of that is actually greater:
- downstream won't be able to rely on this constraint
- the additional complexity to `tox.ini` is probably not worth the expected gain

Reply:
Why don't you write down which version of `ipykernel` is needed to avoid the deprecation warning?

Reply:
Because there is no clear solution in this direction: even if I upgrade `ipykernel` to its current latest version (6.29.5), the same deprecation warning will still be raised in the same test (`table/tests/test_jsviewer.py::test_show_in_notebook_ipydatagrid`). This is because the deprecated code actually lives in some intermediate package layer (`jupyter_client`) that is apparently not caught up with `jupyter_core` right now.

Reply:
Why do we even depend on ipykernel?

Reply:
but that's just for if it's already installed, why are we encouraging people to install it. Oh I give up with astropy's dependencies, it's just a farce.

Reply:
For a deprecation warning, I think it is also fine to just ignore the warning in `pytest`.

@Cadair - I agree the `all` dependency is a bit of a mess: `all` should just add things that add functionality, not things where we provide an interface if they are present. Those should be in `test_all`, as they those interfaces do need to be tested. But I think in fairness package dependencies is simply not a simple problem.

Reply:
@neutrinoceros - yes, makes sense. I certainly think this is a detail that should not hold up the PR.

Reply:
> but that's just for if it's already installed, why are we encouraging people to install it. 

We should start using the new(ish) `[dependency-groups]`. https://peps.python.org/pep-0735/
This aligns much more closely with what we mean / want for things like the jupyter group here.

Reply:
PEP 735 is very nice indeed and I'm adopting it in many small packages I control but I think we should at least wait for pip to support it. tox and uv already do but don't represent what most of us use.
Perhaps an even more important problem with migrating from extras to groups is that extras are a form of public interface, and as such, should be assumed to be already in use, and there's no obvious deprecation path in that space.

Reply:
Yes, please do not forget about users who might be locked in to older versions of things in their env, so we cannot immediately use shiny new things most of the time. Is there a follow up issue about PEP 735?